payu/models/cice.py

```diff
@@ -62,27 +62,63 @@ def set_model_pathnames(self):
         setup_nml = self.ice_in['setup_nml']
 
         res_path = os.path.normpath(setup_nml['restart_dir'])
+        input_dir = setup_nml.get('input_dir', None)
+
+        if input_dir is None:
+            # Default to reading and writing inputs/restarts in-place
+            input_path = res_path
+        else:
+            input_path = os.path.normpath(input_dir)
+
+        # Determine if there is a work input path from grid path
+        grid_nml = self.ice_in['grid_nml']
+        path, _ = os.path.split(grid_nml['grid_file'])
+        if path and not path == '.':
```

Comment:
replace `'.'` with `os.path.curdir` ?

Reply:
`os.path.curdir` is just the platform-agnostic name for `.`, btw.  It doesn't literally mean the current directory.

Yes, I don't think I've used it in the past.

Reply:
Oh right, didn't know that. Ok, I'll add it in.